### ISSUE TITLE
Splunk Streaming Command dict_reader.py attempts to iterate over none

### ISSUE THREAD

**Issue by kkirsche** [splunk/splunk-sdk-python#110]
Hey guys,

For some reason Splunk trys to iterate over none and throws a `TypeError: 'NoneType' object is not iterable` in https://github.com/splunk/splunk-sdk-python/blob/master/splunklib/searchcommands/splunk_csv/dict_reader.py#L38. This is only reproducible in in certain environments it seems (happens in our production distributed cluster(s) but not in local test VMs). This causes the command to still execute but throws an error on the screen cause Splunk sent it no data.

I understand that it _shouldn't_ do this yet still throws the error even though it enriches the data correctly. We were able to duplicate this occurrence with both a custom command and with the `countmatches` example code in our production testbed

**Comment by David-Noble-at-work** [comment]
My code inspection indicates you've discovered a bug and I have what I think is a fix. This:
```python
           try:
                self._fieldnames = self.reader.next()
            except StopIteration:
                pass
```
Should be:
```python
            try:
                self._fieldnames = self.reader.next()
            except StopIteration:
                self._fieldnames = []
```
Given that you've got a repro, I'll ask: Can you give this change a try?

**Comment by kkirsche** [comment]
I'll be able to give it a try on Monday when I'm in our office and can put the code on the server to verify this fixes it. Thanks

**Comment by David-Noble-at-work** [comment]
You're welcome.

**Comment by glennblock** [comment]
Thanks for reporting this Kevin!

**Comment by kkirsche** [comment]
Course. Thanks for such quick help and for directing me where to get this kinda thing reported @glennblock

**Comment by kkirsche** [comment]
@David-Noble-at-work @glennblock The suggested fix to the API code appears to be working on the Splunk 6 search heads. Thanks guys!

**Comment by glennblock** [comment]
w00t!

**Comment by shakeelmohamed** [comment]
Fixed in v1.4.0 which we just released!
